### PR TITLE
add tests with regular tenant or ruleResolver.RulesFor

### PR DESCRIPTION
**Pull request by pdgetrf**
RulesFor was missing test cases using a regular tenant. Added now as promised at a [previous PR](https://github.com/futurewei-cloud/arktos/pull/164#discussion_r403387782)

Also changed the output format of the test to show expected and actual instead of the diff. With the diff output one can't tell what ***should be*** the correct output.

### CODE REVIEW

File: pkg/registry/rbac/validation/rule_test.go [hunk 1]
```diff
@@ -144,7 +187,7 @@ func TestDefaultRuleResolver(t *testing.T) {
 			effectiveRules: nil,
 		},
 		{
-			StaticRoles: staticRoles1,
+			StaticRoles: staticRolesWithSystemTenant,
 			// Same as above but without a namespace. Only cluster rules should apply.
```
Comment (chenqianfzh) [inline pkg/registry/rbac/validation/rule_test.go:191]: You mean "same as above but without a tenant", right?

Comment (pdgetrf) [reply pkg/registry/rbac/validation/rule_test.go:191]: this means same as the cases right above it.

Comment (pdgetrf) [reply pkg/registry/rbac/validation/rule_test.go:191]: will add some clarification.

Comment (pdgetrf) [reply pkg/registry/rbac/validation/rule_test.go:191]: this comment is actually a repeat from previous cases. doesn't add much value so it's now removed.

**Pull request merged**